app/components/signature-pill.js

```diff
@@ -11,84 +11,49 @@ export default class SignaturePillComponent extends Component {
   @service intl;
   @service currentSession;
 
-  @tracked signingHubUrl;
-  @tracked isMarked = false;
-  @tracked isPrepared = false;
-  @tracked hasToBeApproved = false;
-  @tracked hasToBeSigned = false;
-  @tracked isSigned = false;
-  @tracked isRefused = false;
-  @tracked isCancelled = false;
-
-  constructor() {
-    super(...arguments);
-
-    this.loadData.perform();
-  }
+  status = trackedFunction(this, async () => {
+    const signMarkingActivity = await this.args.signMarkingActivity;
+    const signSubcase = await signMarkingActivity.signSubcase;
 
-  get skin() {
-    if (this.isRefused || this.isCancelled) {
-      return "error";
-    } else if (this.signingHubUrl) {
-      return "link"
-    } else {
-      return "ongoing"
-    }
-  }
+    return await (await signSubcase.signFlow).status;
+  });
 
-  get title() {
-    if (this.isCancelled) {
-      return this.intl.t('cancelled');
-    } else if (this.isRefused) {
-      return this.intl.t('refused');
-    } else if (this.isSigned) {
-      return this.intl.t('signed');
-    } else if (this.hasToBeSigned) {
-      return this.intl.t('to-sign');
-    } else if (this.hasToBeApproved) {
-      return this.intl.t('to-approve');
-    } else if (this.isPrepared) {
-      return this.intl.t('sent');
-    } else if (this.isMarked) {
-      return this.intl.t('to-sign');
+  signingHubUrl = trackedFunction(this, async () => {
+    const { SIGNED, REFUSED } = constants.SIGNFLOW_STATUSES;
+    if (!this.status.value || this.status.value.uri === REFUSED) {
+      return null;
     }
-    return "";
-  }
 
-  loadData = task(async () => {
+    const currentUser = this.currentSession.user;
+    const status = this.status.value;
```

Comment:
this assignment is quite unnecessary. Removing it

Reply:
> Also note that after an await, the this context should not be accessed as it could lead to destruction/timing issues.
from https://ember-resources.pages.dev/funcs/util_function.trackedFunction
That is why I created this variable

Reply:
Thinking about this, I realised my approach probably won't solve the problem since this is a reference to the same object `this` would be pointing to, which has the same problems as using `this` directly has.